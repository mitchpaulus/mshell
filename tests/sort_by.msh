[3 1 2 0] (> (-1) (1) iff) sortByCmp str wl
[3 1 2 0] (<= (-1) (1) iff) sortByCmp str wl
[] (> (-1) (1) iff) sortByCmp str wl

# Test multiple levels of sorting
[
    [1 6]
    [0 5]
    [2 4]
    [0 3]
]
(b! a! @a :1: @b :1: <= (-1) (1) iff) sortByCmp
(b! a! @a :0: @b :0: <= (-1) (1) iff) sortByCmp
str wl


# Test version sorting

def compareAndWrite (int str str -- )
    b! a! $"'{@a}' '{@b}' " w @a @b versionSortCmp = str wl
end

0 "a" "a" compareAndWrite
0 "" ""  compareAndWrite
-1 "" "a"  compareAndWrite
1 "a" "" compareAndWrite
-1 "a" "b"  compareAndWrite
1 "b" "a" compareAndWrite
-1 "A" "a" compareAndWrite
1 "a" "A" compareAndWrite
-1 "1" "02" compareAndWrite
1 "02" "2" compareAndWrite
1 "10" "2" compareAndWrite
