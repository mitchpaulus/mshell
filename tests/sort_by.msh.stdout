[3 2 1 0]
[0 1 2 3]
[]
[[0 3] [0 5] [1 6] [2 4]]
'a' 'a' true
'' '' true
'' 'a' true
'a' '' true
'a' 'b' true
'b' 'a' true
'A' 'a' true
'a' 'A' true
'1' '02' true
'02' '2' true
'10' '2' true
